From 44ccdf8c34671a47a13ecef704178f545d4a8770 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Tue, 15 Aug 2017 22:03:35 +0800
Subject: [PATCH 49/80] drm/sun4i: add support for A64 DE2 mixers

The DE2 of A64 has two mixers, one have one VI channel and 3 UI channel,
the other have only one VI and one UI channel.

Add support for them.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/gpu/drm/sun4i/sun8i_mixer.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/gpu/drm/sun4i/sun8i_mixer.c b/drivers/gpu/drm/sun4i/sun8i_mixer.c
index d658a3a8159a..2a69e0953726 100644
--- a/drivers/gpu/drm/sun4i/sun8i_mixer.c
+++ b/drivers/gpu/drm/sun4i/sun8i_mixer.c
@@ -400,6 +400,16 @@ static const struct sun8i_mixer_cfg sun8i_h3_mixer1_cfg = {
 	.ui_num = 1,
 };
 
+static const struct sun8i_mixer_cfg sun50i_a64_mixer0_cfg = {
+	.vi_num = 1,
+	.ui_num = 3,
+};
+
+static const struct sun8i_mixer_cfg sun50i_a64_mixer1_cfg = {
+	.vi_num = 1,
+	.ui_num = 1,
+};
+
 static const struct of_device_id sun8i_mixer_of_table[] = {
 	{
 		.compatible = "allwinner,sun8i-v3s-de2-mixer",
@@ -413,6 +423,14 @@ static const struct of_device_id sun8i_mixer_of_table[] = {
 		.compatible = "allwinner,sun8i-h3-de2-mixer1",
 		.data = &sun8i_h3_mixer1_cfg
 	},
+	{
+		.compatible = "allwinner,sun50i-a64-de2-mixer0",
+		.data = &sun50i_a64_mixer0_cfg
+	},
+	{
+		.compatible = "allwinner,sun50i-a64-de2-mixer1",
+		.data = &sun50i_a64_mixer1_cfg
+	},
 	{ }
 };
 MODULE_DEVICE_TABLE(of, sun8i_mixer_of_table);
-- 
2.13.6

